### PR TITLE
fix(api): preserve order when merging semantic and fulltext results

### LINKED ISSUE #2095: position invalide outil préavis de licenciement navigation sticky
Quand j'arrive sur cette page de modèle, on me propose deux outils sur le côté dont le préavis de licenciement. 
Le préavis de licenciement ne devrait pas apparaître. d'ailleurs quand je tape le modèle dans la barre de recherche, l'outil de préavis de licenciement n'apparait pas dans les résultats. Bizarre. 
- page modèle : https://master-code-travail.dev.fabrique.social.gouv.fr/modeles-de-courriers/rupture-de-periode-dessai-a-linitiative-de-lemployeur?q=Rupture%20de%20p%C3%A9riode%20d%E2%80%99essai%20%C3%A0%20l%27initiative%20de%20l%27employeur
- page recherche : https://master-code-travail.dev.fabrique.social.gouv.fr/recherche?q=Rupture%20de%20p%C3%A9riode%20d%E2%80%99essai%2C%20%C3%A0%20l%E2%80%99initiative%20de%20l%E2%80%99employeur

A noter que je n'ai pas ce problème sur la prod qui fonctionne parfaitement : https://code.travail.gouv.fr/modeles-de-courriers/rupture-de-periode-dessai-a-linitiative-de-lemployeur?q=Mod%C3%A8le%20:%20Rupture%20de%20p%C3%A9riode%20d%E2%80%99essai%20%C3%A0%20l%27initiative%20de%20l%27employeur

### PR DESCRIPTION
**Pull request by lionelB**
this PR 
- update mock data 
- preverse order when merging fulltext results with semantic results

fix #2095, #2152

### CODE REVIEW

**Review by revolunet — APPROVED** [review]

**Comment by lionelB** [comment]
wait before merging, need more testing

**Review by revolunet — APPROVED** [review]

File: packages/code-du-travail-api/src/server/routes/search/utils.js [hunk 1]
```diff
@@ -16,34 +16,29 @@ const merge = (res1, res2, max_result) => {
...
-  Object.values(
-    arr.reduce(
-      (values, current) => ({
-        ...values,
-        [current._source.source + "/" + current._source.slug]: current
-      }),
-      {}
-    )
-  );
+const removeDuplicate = (arr, predicate = predicateFn) =>
```
Comment (UnbearableBear) [inline packages/code-du-travail-api/src/server/routes/search/utils.js:21]: On a vraiment besoin d'un `flatMap` ?
Juste un filter avec un some dedans ça me parait un peu plus explicite pour filtrer si on n'a pas de soucis de sous tableaux !
```
arr.filter(
  (item, index) =>
    !arr.some(
      (subItem, subIndex) =>
        subIndex < index && predicate(item) === predicate(subItem)
    )
);
```

Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/search/utils.js:21]: filter is ok but I found the `some` syntax very hard to read

Comment (UnbearableBear) [reply packages/code-du-travail-api/src/server/routes/search/utils.js:21]: It's basically the same than `.find` imho

**Pull request merged**